streams/shared/src/main/scala/zio/stream/StreamEffect.scala

```diff
@@ -295,31 +296,54 @@ private[stream] class StreamEffect[-R, +E, +A](val processEffect: ZManaged[R, E,
           try {
             pull().toInt
           } catch {
-            case StreamEffect.End        => -1
-            case StreamEffect.Failure(e) => throw e.asInstanceOf[E]
+            case PullEffect.End        => -1
+            case PullEffect.Failure(e) => throw e.asInstanceOf[E]
           }
       }
     } yield javaStream
 }
 
 private[stream] object StreamEffect extends Serializable {
 
-  case class Failure[E](e: E) extends Throwable(e.toString, null, true, false) {
-    override def fillInStackTrace() = this
-  }
+  object PullEffect {
```

Comment:
Again wary of more concepts here. Let's drop the objects and just keep everything on the `StreamEffect` companion. It's all private anyway.

Reply:
The reasoning behind this was to mirror the `ZStream` code organization. Otherwise, I don't have any feelings for this.

Reply:
Done.